File: app/src/main/java/maw/org/wedding/map/Fetching/PlacePhotoFetcher.java
package maw.org.wedding.map.Fetching;

import android.content.Context;

import com.squareup.picasso.Callback;
import com.squareup.picasso.Picasso;

import maw.org.wedding.map.PhotoRequestListener;

public class PlacePhotoFetcher {

    public void fetchPhotoForReference(Context context, String photoReference, String apiKey, String maxWidth, final PhotoRequestListener listener) {
        String baseURL = "https://maps.googleapis.com/maps/api/place/photo?";

        Integer width = Integer.parseInt(maxWidth);

        String newWidth = String.valueOf(width/3);

        baseURL += "key=" + apiKey;
        baseURL += "&photoreference=" + photoReference;
        baseURL += "&maxwidth=" + newWidth;

        final String finalBaseURL = baseURL;
        Picasso.with(context).load(baseURL).fetch(new Callback() {
            @Override
            public void onSuccess() {
                listener.onSuccess(finalBaseURL);
            }

            @Override
            public void onError() {
                listener.onFailure();
            }
        });

    }
}

File: app/src/main/kotlin/maw/org/wedding/map/Fetching/PlacePhotoFetcher.kt
package maw.org.wedding.map.Fetching

import android.content.Context

import com.squareup.picasso.Callback
import com.squareup.picasso.Picasso

import maw.org.wedding.map.PhotoRequestListener

class PlacePhotoFetcher {

    fun fetchPhotoForReference(context: Context, photoReference: String, apiKey: String, maxWidth: String, listener: PhotoRequestListener) {
        var baseURL = "https://maps.googleapis.com/maps/api/place/photo?"

        val width = Integer.parseInt(maxWidth)

        val newWidth = (width / 3).toString()

        baseURL += "key=" + apiKey
        baseURL += "&photoreference=" + photoReference
        baseURL += "&maxwidth=" + newWidth

        val finalBaseURL = baseURL
        Picasso.with(context).load(baseURL).fetch(object : Callback {
            override fun onSuccess() {
                listener.onSuccess(finalBaseURL)
            }

            override fun onError() {
                listener.onFailure()
            }
        })

    }
}
